pkg/common/cliprinter/cliprinter.go

```diff
@@ -0,0 +1,131 @@
+package cliprinter
+
+import (
+	"io"
+	"os"
+	"sync"
+
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorjson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorpretty"
+	"github.com/spiffe/spire/pkg/common/cliprinter/protojson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/protopretty"
+	"github.com/spiffe/spire/pkg/common/cliprinter/structjson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/structpretty"
+	"google.golang.org/protobuf/proto"
+)
+
+type Printer interface {
+	PrintError(error)
+	PrintProto(...proto.Message)
+	PrintStruct(...interface{})
+}
+
+type DefaultPrinter struct {
```

Comment:
nitpick: the `DefaultPrinter` name was a little confusing since this type is returned from both NewDefaultPrinter() and NewPrinter()